### PR TITLE
refactor(WalletAPI): AND-1300 Inject WalletAPI in most places

### PR DESCRIPTION
**Pull request by westonal**

### CODE REVIEW

File: wallet/src/main/java/info/blockchain/wallet/payload/PayloadManager.java [hunk 1]
```diff
@@ -94,7 +95,7 @@ private PayloadManager() {
     }
 
     private void init() {
-        walletApi = new WalletApi();
+        walletApi = WalletApiAccess.INSTANCE.getWalletApi();
```
Comment (westonal) [inline wallet/src/main/java/info/blockchain/wallet/payload/PayloadManager.java]: There are two places that can't yet be injected. This is one.

File: wallet/src/main/java/info/blockchain/wallet/payload/data/LegacyAddress.java [hunk 2]
```diff
@@ -180,8 +181,8 @@ public static LegacyAddress fromECKey(ECKey ecKey) throws Exception {
     }
 
     private static ECKey getRandomECKey() throws Exception {
-
-        Call<ResponseBody> call = new WalletApi().getRandomBytesCall();
+        // TODO: Unused apart from in tests, remove as part of AND-1194
+        Call<ResponseBody> call = WalletApiAccess.INSTANCE.getWalletApi().getRandomBytesCall();
```
Comment (westonal) [inline wallet/src/main/java/info/blockchain/wallet/payload/data/LegacyAddress.java]: There are two places that can't yet be injected. This is the second. Covered by AND-1194.

File: wallet/src/main/java/info/blockchain/wallet/api/WalletApiEndpoints.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import io.reactivex.Observable
+import okhttp3.ResponseBody
+import retrofit2.Call
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface WalletApiEndpoints {
+
+    @GET("v2/randombytes")
+    fun getRandomBytesCall(
```
[the remark this answers is not in the document]
Comment (westonal) [reply wallet/src/main/java/info/blockchain/wallet/api/WalletApiEndpoints.kt]: Yeah, maybe covered by AND-1194 I didn't realize, but it was useful to split it out upfront still I think.

[the remark this answers is not in the document]
Comment (ditn) [reply wallet/src/main/java/info/blockchain/wallet/api/WalletApiEndpoints.kt]: Cool, makes sense.

File: wallet/src/test/java/info/blockchain/wallet/MockedResponseTest.java [hunk 4]
```diff
@@ -2,99 +2,99 @@
...
         });
-        RxJavaPlugins.setInitComputationSchedulerHandler(new Function<Callable<Scheduler>, Scheduler>() {
+        RxJavaPlugins.setComputationSchedulerHandler(new Function<Scheduler, Scheduler>() {
             @Override
-            public Scheduler apply(Callable<Scheduler> schedulerCallable) {
+            public Scheduler apply(Scheduler schedulerCallable) {
                 return TrampolineScheduler.instance();
             }
         });
-        RxJavaPlugins.setInitNewThreadSchedulerHandler(new Function<Callable<Scheduler>, Scheduler>() {
```
Comment (westonal) [inline wallet/src/test/java/info/blockchain/wallet/MockedResponseTest.java]: These `setInit...` calls could be responsible for many of the flakey tests.

**Pull request merged**